dev/benchmarks/microbenchmarks/lib/building/const_vs_non_const_bench.dart

```diff
@@ -0,0 +1,230 @@
+import 'package:flutter_test/flutter_test.dart';
+
+import '../common.dart';
+
+const Duration kBenchmarkTime = Duration(seconds: 15);
+
+Future<List<double>> runBuildBenchmark(ValueGetter<Widget> buildApp) async {
+  assert(false, "Don't run benchmarks in debug mode! Use 'flutter run --release'.");
+
+  // We control the framePolicy below to prevent us from scheduling frames in
+  // the engine, so that the engine does not interfere with our timings.
+  final LiveTestWidgetsFlutterBinding binding = TestWidgetsFlutterBinding.ensureInitialized() as LiveTestWidgetsFlutterBinding;
+
+  final Stopwatch watch = Stopwatch();
+  int iterations = 0;
+  final List<double> values = <double>[];
+
+  await benchmarkWidgets((WidgetTester tester) async {
+    await tester.pumpWidget(buildApp());
+    await tester.pump();
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+
+    // Cannot use expects/asserts here since this is running outside of a test
+    // in release mode.
+    final int numberOfSizedBoxes = find.byType(SizedBox).evaluate().length;
+    if (numberOfSizedBoxes != 30) {
+      throw StateError('Expected 30 SizedBox widgets, but only found $numberOfSizedBoxes.');
+    }
+    if (find.text('testToken').evaluate().length != 1) {
+      throw StateError('Did not find expected leaf widget.');
+    }
+
+    final Element rootWidget = tester.element(find.byKey(rootKey));
+    Duration elapsed = Duration.zero;
+
+    final LiveTestWidgetsFlutterBindingFramePolicy defaultPolicy = binding.framePolicy;
+    binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.benchmark;
+
+    while (elapsed < kBenchmarkTime) {
+      watch.reset();
+      watch.start();
+      rootWidget.markNeedsBuild();
+      await tester.pumpBenchmark(Duration(milliseconds: iterations * 16));
+      watch.stop();
+      iterations += 1;
+      elapsed += Duration(microseconds: watch.elapsedMicroseconds);
+      values.add(watch.elapsedMicroseconds.toDouble());
+      print(values.last);
+    }
+
+    binding.framePolicy = defaultPolicy;
+  });
+  return values;
+}
+
+double calculateMean(List<double> values) {
+  return values.reduce((double x, double y) => x + y) / values.length;
+}
+
+Future<void> main() async {
+  final BenchmarkResultPrinter printer = BenchmarkResultPrinter();
+  final double constMean = calculateMean(await runBuildBenchmark(() => ConstApp(key: rootKey)));
+  final double nonConstMean = calculateMean(await runBuildBenchmark(() => NonConstApp(key: rootKey)));
+  printer.addResult(
+    description: 'const app build',
+    value: constMean,
+    unit: 'µs per iteration',
+    name: 'const_app_build_iteration',
+  );
+  printer.addResult(
+    description: 'non-const app build',
+    value: nonConstMean,
+    unit: 'µs per iteration',
+    name: 'non_const_app_build_iteration',
+  );
+  printer.addResult(
+    description: 'const speed-up (vs. non-const)',
+    value: (1.0 - (constMean / nonConstMean)) * 100,
```

Comment:
Hmm if an operation used to take 10s but now takes 5s I thought that should be called a 100% speed up? But with this formula that would only be a 50% speedup?

Reply:
Done. Can you check my updated math?